### PR TITLE
check to make sure map coords are a number

### LINKED ISSUE #867: Selecting Location tab in Hotspot Settings fails when there is no asserted location
Steps to reproduce:
1. Add a Hotspot using the onboarding flow
2. Set a location later
3. go to Hotspot details, tap settings
4. Tap Update Settings
5. select Update Location tab
6. crash

Has happened to several users on Android, iOS but the devs are unable to reproduce. Will wait for more crash logs to come through to see if we can dig deeper.

### PR DESCRIPTION
**Pull request by tyler-whitman**
hopefully fixes #867

### CODE REVIEW

File: src/utils/mapUtils.ts [hunk 1]
```diff
@@ -27,7 +28,7 @@ export const findBounds = (
   coords: number[][],
   paddingBottom = 250,
 ): MapBounds | undefined => {
-  if (coords.length === 0) {
+  if (coords.length === 0 || !isNumber(coords[0]) || !isNumber(coords[1])) {
```
Comment (matthewcarlreetz) [inline src/utils/mapUtils.ts:31]: Wouldn't both `coords[0]` and `coords[1]` be arrays here?

File: src/components/Map.tsx [hunk 2]
```diff
@@ -102,7 +103,12 @@ const Map = ({
   }, [onMapMoved])
 
   const centerUserLocation = useCallback(() => {
-    const hasCoords = userCoords && userCoords.longitude && userCoords.latitude
+    const hasCoords =
```
Comment (matthewcarlreetz) [inline src/components/Map.tsx:106]: Null and undefined aren't numbers so I think we can shorten this to 
```ts
 const hasCoords =
      userCoords &&
      isNumber(userCoords.longitude) &&
      isNumber(userCoords.latitude
```

File: src/components/Map.tsx [hunk 3]
```diff
@@ -125,7 +131,13 @@ const Map = ({
   )
 
   useEffect(() => {
-    if (!showUserLocation || !userCoords.latitude || !userCoords.longitude)
+    if (
```
Comment (matthewcarlreetz) [inline src/components/Map.tsx:134]: I think we could do the same here...
```ts
   if (
      !showUserLocation ||
      !isNumber(userCoords.latitude) ||
      !isNumber(userCoords.longitude)
    )
      return
```

File: src/components/Map.tsx [hunk 4]
```diff
@@ -195,13 +207,19 @@ const Map = ({
     return findBounds(boundsLocations, cameraBottomOffset)
   }, [mapCenter, cameraBottomOffset, selectedHex, selectedHotspot, witnesses])
 
-  const defaultCameraSettings = useMemo(
-    () => ({
+  const defaultCameraSettings = useMemo(() => {
+    const centerCoordinate =
```
Comment (matthewcarlreetz) [inline src/components/Map.tsx:211]:
```ts
    const centerCoordinate =
      mapCenter?.length === 2 &&
      isNumber(mapCenter[0]) &&
      isNumber(mapCenter[1])
        ? mapCenter
        : defaultLngLat
```

File: src/utils/mapUtils.ts (hunk 1 above)
Comment (tyler-whitman) [reply src/utils/mapUtils.ts:31]: Good catch, moved the check down to check the actual numbers

File: src/components/Map.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/components/Map.tsx:106]: updated

File: src/components/Map.tsx (hunk 3 above)
Comment (tyler-whitman) [reply src/components/Map.tsx:132]: updated

File: src/components/Map.tsx (hunk 4 above)
Comment (tyler-whitman) [reply src/components/Map.tsx:207]: updated

**Review by matthewcarlreetz — APPROVED** [review]
